SwiftAA/Angles.swift

```diff
@@ -11,14 +11,34 @@ import Foundation
 /// The Degree is a unit of angle.
 /// Degree structs conform to SwiftAA Numeric type protocol.
 public struct Degree: NumericType, CustomStringConvertible {
-    /// The Degree value
-    public let value: Double
     
+    public var magnitude: Double
+    
+    public static func -=(lhs: inout Degree, rhs: Degree) {
+        lhs.value = lhs.value - rhs.value
+    }
+    
+    public static func +=(lhs: inout Degree, rhs: Degree) {
+        lhs.value = rhs.value + lhs.value
+    }
+    
+    public static func *(lhs: Degree, rhs: Degree) -> Degree {
+        return Degree(lhs.value * rhs.value)
+    }
```

Comment:
I would suggest to fall back to the generic parent NumericType for cases like this. Otherwise this will be a problem with no end. I don't have an Xcode to test though.